BASE/src/Microsoft.ApplicationInsights/Extensibility/TelemetryConfiguration.cs

```diff
@@ -94,7 +95,8 @@ public TelemetryConfiguration(string instrumentationKey, ITelemetryChannel chann
         {
             this.instrumentationKey = instrumentationKey ?? throw new ArgumentNullException(nameof(instrumentationKey));
 
-            SetTelemetryChannelEndpoint(channel, this.EndpointContainer.FormattedIngestionEndpoint, force: true);
+            var ingestionEndpoint = this.EndpointContainer.GetFormattedIngestionEndpoint(enableAAD: this.CredentialEnvelope != null);
```

Comment:
can users call TelemetryConfig.SetCredentiaEnveloper afterwards and it gets picked up?

Reply:
Yeah, i copied the design from Connection String.

Regardless of which is set first, TelemetryChannel or SetCredential, the endpoint will be re-evaluated.
This is because AAD uses ingestion v2.1. 
In a future update, this won't be needed when we transition the whole SDK to v2.1 next semester.

Reply:
I'm writing unit tests to verify this behavior right now.